src/test/basics/Buffer_test.cpp

```diff
@@ -27,15 +29,6 @@ namespace test {
 
 struct Buffer_test : beast::unit_test::suite
 {
-    bool
-    sane(Buffer const& b) const
-    {
-        if (b.size() == 0)
-            return b.data() == nullptr;
```

Comment:
I'd like to clarify that the new code does not adhere to the old behavior. When `size_ == 0`, `data_` will point to garbage data in the proposed code, whereas in the existing code, `data_ == nullptr`

Reply:
Yes, good eye. That is an implementation detail, which the tests checked for, and that's what's happening here.

Having the internal `data_` pointer always pointing to valid memory (what's known as an [invariant](https://en.wikipedia.org/wiki/Class_invariant)) and never to `nullptr` simplifies other parts of the code (always a good thing), and imposes no additional cost.

Reply:
okay thanks for the clarification.

Is the invariant regarding the non-nullness of the `data_` pointer widely held across the codebase? I'm wondering if stating it as an assert or a comment would help.

Reply:
The invariant is internal to the class. External code is unaware of it.